java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -634,6 +635,26 @@ private void executeBinaryExpression(Tree tree) {
     programState = unstackBinary.state;
     SymbolicValue symbolicValue = constraintManager.createSymbolicValue(tree);
     symbolicValue.computedFrom(unstackBinary.values);
+    if(tree.is(Tree.Kind.PLUS)) {
+      BinaryExpressionTree bt = (BinaryExpressionTree) tree;
+      if (bt.leftOperand().symbolType().is("java.lang.String")) {
+        Constraint leftConstraint = programState.getConstraint(unstackBinary.values.get(1));
+        if (leftConstraint != null && !leftConstraint.isNull()) {
```

Comment:
The processing of the constraint is exactly the same for the left and the right constraints. It would be easier to handle them with a common method.